resources/splunk_app.rb

```diff
@@ -136,3 +148,41 @@ def config_scope
     apply_config
   end
 end
+
+class PackagedApp < SplunkApp
+  property :source_url, String, required: true
+
+  resource_name :splunk_app_package
+
+  def package_path
+    return @package_path if @package_path
+    file_cache = Pathname.new(Chef::Config['file_cache_path'])
+    @package_path = file_cache.join(CernerSplunk::PathHelpers.filename_from_url(source_url).gsub(/.spl$/, '.tgz'))
+  end
+
+  action :install do
+    return unless !version || changed?(:version)
+
+    remote_file package_path.to_s do
```

Comment:
Also, should the validation instead be on that the package version matches the desired version (since that is a parameter)? I could see it being a strict match to the desired version to make sure the downloaded app is exactly what is expected, or simply checking that the downloaded app is a release version if a release version is given. The version validation, I feel, are between the package's version and the version property, rather than against the installed version.